### PR TITLE
📝(project) update ralph v4 upgrade guide and changelog

### PR DESCRIPTION
**Pull request by SergioSim**
## Purpose

We want to keep our changelog and upgrade guide up to date with the latest changes made to Ralph's backends.

## Proposal

- [x] Update changelog and upgrade guide

### CODE REVIEW

File: UPGRADE.md
Comment (wilbrdt) [inline UPGRADE.md:1]: As we introduced the `LRS` backends in v4, I think we should have a section `Upgrade Ralph LRS usage` explaining that the `DATABASE` backends are now `LRS` backends, and that the corresponding environment variables should be configured. What else could be added to this section?

File: src/ralph/backends/data/swift.py [hunk 1]
```diff
@@ -51,16 +51,18 @@ class Config(BaseSettingsConfig):
...
 
     AUTH_URL: str = "https://auth.cloud.ovh.net/"
-    USERNAME: Optional[str] = None
-    PASSWORD: Optional[str] = None
+    DEFAULT_CONTAINER: Optional[str] = None
     IDENTITY_API_VERSION: str = "3"
-    TENANT_ID: Optional[str] = None
-    TENANT_NAME: Optional[str] = None
+    OBJECT_STORAGE_URL: Optional[str] = None
+    PASSWORD: Optional[str] = None
```
Comment (wilbrdt) [inline src/ralph/backends/data/swift.py:57]: I don't think these changes are supposed to be here?

Comment (SergioSim) [reply src/ralph/backends/data/swift.py:57]: Good point) Thanks) Moved it to a separate commit)

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,24 +39,28 @@ methods under the unified `lrs` backend interface [BC]
...
 have an authority field matching that of the user
 - CLI: change `push` to `write` and `fetch` to `read` [BC]
-- Upgrade `bcrypt` to `4.1.1`
+- CLI: change `-c --chunk-size` option to `-s --chunk-size` [BC]
+- CLI: change websocket backend name `-b ws` to `-b async_ws` along with it's
+  uri option `--ws-uri` to `--async-ws-uri` [BC]
+- Upgrade `bcrypt` to `>=4.0.0`
 - Upgrade `fastapi` to `0.105.0`
 - Upgrade `sentry_sdk` to `1.38.0`
 - Upgrade `uvicorn` to `0.24.0.post1`
```
[the remark this answers is not in the document]
Comment (SergioSim) [reply CHANGELOG.md]: Good idea) Thanks) Was trying to limit changes to facilitate review) Updated with grouping similar sections)

File: CHANGELOG.md [hunk 3]
```diff
@@ -70,6 +79,7 @@ have an authority field matching that of the user
 - `school`, `course`, `module` context extensions in Edx to xAPI base converter
 - `name` field in `VideoActivity` xAPI model mistakenly used in `video` profile
 - drop support for python 3.7
```
[the remark this answers is not in the document]
Comment (SergioSim) [reply CHANGELOG.md]: Do we consider the drop of python 3.7 support as a change or a removal?

File: CHANGELOG.md [hunk 4]
```diff
@@ -34,24 +39,28 @@ methods under the unified `lrs` backend interface [BC]
...
   in the video and virtual classroom profiles. [BC]
-- Backends: `LRSHTTP` methods must not be used in `asyncio` events loop (BC)
 - Add variable to override PVC name in arnold deployment
-- Backends: add `max_statements` option to `AsyncLRSHTTP`
 - API: Incoming statements are enriched with `id`, `timestamp`, `stored`
   and `authority`
 - Backends: update `statementId` and `voidedStatementId` to snake_case,
   with camelCase alias, in `LRSStatementsQuery`
 - API: Add `RALPH_LRS_RESTRICT_BY_AUTHORITY` option making `?mine=True`
   implicit
```
[the remark this answers is not in the document]
Comment (SergioSim) [reply CHANGELOG.md:38]: It's indeed easier to read when entries are grouped) Thanks) Updated)

File: CHANGELOG.md [hunk 5]
```diff
@@ -70,6 +79,7 @@ have an authority field matching that of the user
 - `school`, `course`, `module` context extensions in Edx to xAPI base converter
 - `name` field in `VideoActivity` xAPI model mistakenly used in `video` profile
 - drop support for python 3.7
+- `DEFAULT_BACKEND_CHUNK_SIZE` environment variable
```
[the remark this answers is not in the document]
Comment (SergioSim) [reply CHANGELOG.md]: Good point) Thanks) Updated this with:
```
CLI: Remove `DEFAULT_BACKEND_CHUNK_SIZE` environment variable configuration
```

File: UPGRADE.md [hunk 6]
```diff
@@ -71,6 +71,49 @@ For example here is a valid `auth.json` file:
...
+    | clickhouse        | RALPH_BACKENDS__DATA__CLICKHOUSE__READ_CHUNK_SIZE=500 |
+    | fs                | RALPH_BACKENDS__DATA__FS__READ_CHUNK_SIZE=4096        |
+    | ldp               | RALPH_BACKENDS__DATA__LDP__READ_CHUNK_SIZE=4096       |
+    | s3                | RALPH_BACKENDS__DATA__S3__READ_CHUNK_SIZE=4096        |
+    | swift             | RALPH_BACKENDS__DATA__SWIFT__READ_CHUNK_SIZE=4096     |
+
+- The `ralph push` command changed to `ralph write`
+  - The `-c --chunk-size` option changed to `-s --chunk-size`
+  - The `DEFAULT_BACKEND_CHUNK_SIZE` environment variable configuration is removed in
+    favor of allowing each backend define their own defaults:
```
[the remark this answers is not in the document]
Comment (SergioSim) [reply UPGRADE.md]: Well spotted) Thanks) Updated)

File: UPGRADE.md
Comment (SergioSim) [reply UPGRADE.md:1]: Well spotted) Thanks) Added a mention about changes of environment variables and the interface change)

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (wilbrdt) [reply CHANGELOG.md]: We could put it in `removal` indeed

**Review by wilbrdt — APPROVED** [review]
LGTM! :tada:

**Pull request merged**